tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import {Client} from 'app/api';
+import {mountWithTheme} from 'sentry-test/enzyme';
+import PluginDetailedView from 'app/views/organizationIntegrations/pluginDetailedView';
+
+const mockResponse = mocks => {
+  mocks.forEach(([url, body]) => {
+    return Client.addMockResponse({
+      url,
+      body,
+    });
+  });
+};
+
+describe('PluginDetailedView', function() {
+  const org = TestStubs.Organization();
+  const routerContext = TestStubs.routerContext();
+  let wrapper;
+
+  beforeEach(() => {
+    Client.clearMockResponses();
+
+    mockResponse([
+      [
+        `/organizations/${org.slug}/plugins/configs/?plugins=pagerduty`,
+        [
+          {
+            status: 'unknown',
+            description: 'Send alerts to PagerDuty.',
+            isTestable: true,
+            isHidden: false,
+            hasConfiguration: true,
+            features: [],
+            shortName: 'PagerDuty',
+            id: 'pagerduty',
+            assets: [],
+            featureDescriptions: [],
+            name: 'PagerDuty',
+            author: {url: 'https://github.com/getsentry/sentry', name: 'Sentry Team'},
+            contexts: [],
+            doc: '',
+            resourceLinks: [
+              {url: 'https://github.com/getsentry/sentry/issues', title: 'Report Issue'},
+              {
+                url: 'https://github.com/getsentry/sentry/tree/master/src/sentry_plugins',
+                title: 'View Source',
+              },
+            ],
+            slug: 'pagerduty',
+            projectList: [
+              {
+                projectId: 2,
+                configured: true,
+                enabled: true,
+                projectSlug: 'javascript',
+
+                projectPlatform: 'javascript',
+                projectName: 'JavaScript',
+              },
+            ],
+            version: '10.1.0.dev0',
+            canDisable: true,
+            type: 'notification',
+            metadata: {},
+          },
+        ],
+      ],
+    ]);
+
+    wrapper = mountWithTheme(
+      <PluginDetailedView
+        params={{integrationSlug: 'pagerduty', orgId: org.slug}}
+        location={{query: {}}}
+      />,
+      routerContext
+    );
+  });
+  it('shows the Integration name and install status', async function() {
+    expect(wrapper.find('Name').props().children).toEqual('PagerDuty (Legacy)');
+    expect(wrapper.find('IntegrationStatus').props().status).toEqual('Installed');
+  });
+  it('shows the Add to Project button', async function() {
+    expect(wrapper.find('AddButton').props().disabled).toEqual(false);
+    expect(wrapper.find('AddButton').props().children).toEqual('Add to Project');
```

Comment:
@NisanthanNanthakumar Is there no way to test that the modal opens?

Reply:
@scefali unfortunately, it's not a function passed into props that I can mock with Jest. Also the component is out of the scope of the `wrapper` dom, so I can't see if it shows up.

Reply:
@NisanthanNanthakumar You can mock things other than props with jest. But you might need to change the import of the modal like such:
```
import * as modal from 'app/actionCreators/modal';

...

modal.openModal(
```
Then you can mock `openModel` on the `modal` object.